### ISSUE TITLE
Ensure ZeroDataStateNotifications work correctly for GA4 dashboard view.

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#6838]
## Feature Description

At present the `ZeroDataStateNotifications` component has a couple of hardwired references to `analytics` which may not work correctly for the GA4 dashboard view.

One of these is the check for whether `analytics` is shareable. At present if UA is not connected, `analytics` is not shareable, although `analytics-4` may be, which means `ZeroDataStateNotifications` may not behave as expected on the GA4 dashboard view. The approach taken in https://github.com/google/site-kit-wp/issues/6824 may be relevant here.

The second is a check for whether `analytics` is recoverable. It's not clear whether this is a valid check to make on the GA4 dashboard view. Additionally, it raises a question of whether it's possible for the `analytics-4` module to be put into a recoverable state.

Both of these issues need to be investigated and addressed as necessary.

https://github.com/google/site-kit-wp/blob/054ffd4664328ff675e6231bca140bbdfe6e4c90/assets/js/components/notifications/ZeroDataStateNotifications/index.js#L52-L58

https://github.com/google/site-kit-wp/blob/054ffd4664328ff675e6231bca140bbdfe6e4c90/assets/js/components/notifications/ZeroDataStateNotifications/index.js#L66-L79

### Update: A separate issue has been raised which will deal with recoverable modules: https://github.com/google/site-kit-wp/issues/6861

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

- In the scenario where `ga4Reporting` is enabled, GA4 is connected but UA is not, Analytics is shared, and the user is on the view-only Dashboard:
  - The Gathering Data and Zero Data notifications for Analytics should be displayed as expected when the `analytics-4` module is in a gathering or zero data state.

## Implementation Brief

Within `assets/js/components/notifications/ZeroDataStateNotifications/index.js`:

- Update the call to `canViewSharedModule( 'analytics' )` to pass in `analytics-4` instead of `analytics` when `isGA4DashboardView` is true.
https://github.com/google/site-kit-wp/blob/ce3938d94f4e809d6f81ed6b939112e4a7dbea16/assets/js/components/notifications/ZeroDataStateNotifications/index.js#L57

### Test Coverage

* No new tests needed.

## QA Brief

* Ensure the `ga4Reporting` feature flag is enabled.
* Connect a site with zero data (especially zero analytics data). Note: this should be zero data and **not** gathering data. Connect _only_ GA4, not a UA property, ensuring the GA4 dashboard is used.
* Enable Dashboard sharing and share Analytics with another user.
* Sign in as the user who can view the view-only, shared Dashboard. View the main Site Kit Dashboard.
* You should see the zero data notification at the top of the page, even when only GA4 is connected.

## Changelog entry

* Fix zero data notifications on GA4 dashboard.

**Comment by tofumatt** [comment]
IB ✅

**Comment by aaemnnosttv** [comment]
@tofumatt @wpdarren This might be difficult to QA before a few other things are done due to some other changes that have landed which have caused different problems on the view only dashboard.

**Comment by techanvil** [comment]
Yup, this is marked as being blocked by #6687 for this reason, not sure if there are other impacting issues.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

We're unable to test this until 6824 is merged.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@jimmymadon as per our conversation.  

When I go to the view only dashboard, the zero data banner isn't appearing as expected. 

To confirm this is with a site that is set up for GA4 only (no UA is setup) and is on zero data state. 

I see an error in the console which is:

`Googlesitekit-api-a605e8ce718042ee5433.js:1 Google Site Kit API Error method:GET datapoint:conversion-events type:modules identifier:analytics-4 error:"Site Kit can’t access the relevant data from Analytics 4 because you haven’t granted all permissions requested during setup.")`

When setting up Site Kit and Analytics the appropriate permission was given. 

![image](https://user-images.githubusercontent.com/73545194/232040292-ebb05413-e8ed-4248-8751-6043107b1e02.png)

![image](https://user-images.githubusercontent.com/73545194/232040424-356201e8-9556-488b-9199-e70cc6d0a2be.png)

**Comment by hussain-t** [comment]
@wpdarren, I'm unable to reproduce the error you have encountered. As a view-only user, I could see the Zero Data State notification and didn't get the `conversion-events` network error. 

<img width="1792" alt="Screenshot 2023-04-17 at 12 41 04 PM" src="https://user-images.githubusercontent.com/24408261/232412545-6fcbf72a-731b-4228-99dd-f7325dba6ef3.png">

Your screenshot showed that your site was in "Gathering Data State" not "Zero Data State". As per the QAB:

>Connect a site with zero data (especially zero analytics data). Note: this should be zero data and not gathering data.

However, the AC mentions fixing this in both states:

>The Gathering Data and Zero Data notifications for Analytics should be displayed as expected when the analytics-4 module is in a gathering or zero data state.

@tofumatt, why was it mentioned to check only in the "Zero Data State"?

**Comment by wpdarren** [comment]
@hussain-t the gathering state is for search console not for Analytics.  I went through this with Jimmy on a call on Friday, and was able to recreate it a few times.  Let me take some time now though to relook at it and see if the issue still exists.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@hussain-t I tested this with a new site, and set up Analytics with an account I know has a GA4 property with zero data. No UA is set up. The GA4 Dashboard view is being loaded.  When I visit the view only dashboard, the gathering state banner appears for Search console, but no zero data state banner appears as you can see from the screencast below. 

The error I reported previously, you're right does no longer appear, but it was appearing consistently when testing on Friday. Maybe something merged later fixed this issue.  I'm still not seeing the zero data banner though as per QAB. 

https://user-images.githubusercontent.com/73545194/232452678-eadd80a5-18f0-4778-9308-2c18b7f42a27.mp4

**Comment by hussain-t** [comment]
@wpdarren, we don't show both notifications one after another. The gathering data state notification precedes the zero data state notification, and it will be displayed if either of the modules is in the gathering data state. The zero data notification won't be displayed even if you dismiss it. Dismissing doesn’t mean the module is removed from the gathering data state. 

I think this is what @tofumatt intended in the QAB:
>Connect a site with zero data (especially zero analytics data). Note: this should be zero data and not gathering data.

Could you test it on a site with only zero data, not gathering data?

**Comment by wpdarren** [comment]
QA Update: ✅ 

**Verified:**

Viewing the main Site Kit Dashboard, the zero data notification at the top of the page, even when only GA4 is connected.

Note: the zero data notification does not appear when Search console is in gathering state.
